airbyte-bootloader/src/test/java/io/airbyte/bootloader/SecretMigratorTest.java

```diff
@@ -0,0 +1,240 @@
+        .withSourceDefinitionId(sourceDefinitionId)
+        .withConfiguration(sourceConfiguration)
+        .withWorkspaceId(workspaceId);
+    final List<SourceConnection> sourceConnections = Lists.newArrayList(sourceConnection);
+    Mockito.when(configPersistence.listConfigs(ConfigSchema.SOURCE_CONNECTION, SourceConnection.class))
+        .thenReturn(sourceConnections);
+
+    final JsonNode destinationSpec = Jsons.jsonNode("destinationSpec");
+    final UUID destinationDefinitionId = UUID.randomUUID();
+    final StandardDestinationDefinition standardDestinationDefinition = new StandardDestinationDefinition()
+        .withDestinationDefinitionId(destinationDefinitionId)
+        .withSpec(
+            new ConnectorSpecification()
+                .withConnectionSpecification(destinationSpec));
+    final Map<UUID, JsonNode> standardDestinationDefinitions = new HashMap<>();
+    standardDestinationDefinitions.put(destinationDefinitionId, standardDestinationDefinition.getSpec().getConnectionSpecification());
+    Mockito.when(configPersistence.listConfigs(ConfigSchema.STANDARD_DESTINATION_DEFINITION, StandardDestinationDefinition.class))
+        .thenReturn(Lists.newArrayList(standardDestinationDefinition));
+
+    final JsonNode destinationConfiguration = Jsons.jsonNode("destinationConfiguration");
+    final DestinationConnection destinationConnection = new DestinationConnection()
+        .withDestinationId(UUID.randomUUID())
+        .withDestinationDefinitionId(destinationDefinitionId)
+        .withConfiguration(destinationConfiguration)
+        .withWorkspaceId(workspaceId);
+    final List<DestinationConnection> destinationConnections = Lists.newArrayList(destinationConnection);
+    Mockito.when(configPersistence.listConfigs(ConfigSchema.DESTINATION_CONNECTION, DestinationConnection.class))
+        .thenReturn(destinationConnections);
+
+    // Mockito.doNothing().when(secretMigrator).migrateDestinations(Mockito.any(), Mockito.any());
+
+    final String path = "Mocked static call source";
+    Mockito.doReturn(Lists.newArrayList(path)).when(secretMigrator).getSecretPath(sourceSpec);
+    Mockito.doReturn(Lists.newArrayList(path)).when(secretMigrator).getAllExplodedPath(sourceConfiguration, path);
+    final String sourceSecret = "sourceSecret";
+    Mockito.doReturn(Optional.of(Jsons.jsonNode(sourceSecret))).when(secretMigrator).getValueForPath(sourceConfiguration, path);
+    Mockito.doReturn(Lists.newArrayList(path)).when(secretMigrator).getSecretPath(destinationSpec);
+    Mockito.doReturn(Lists.newArrayList(path)).when(secretMigrator).getAllExplodedPath(destinationConfiguration, path);
+    final String destinationSecret = "destinationSecret";
+    Mockito.doReturn(Optional.of(Jsons.jsonNode(destinationSecret))).when(secretMigrator).getValueForPath(destinationConfiguration, path);
+
+    Mockito.doReturn(Jsons.jsonNode("sanitized")).when(secretMigrator).replaceAtJsonNode(Mockito.any(), Mockito.any(), Mockito.any());
```

Comment:
@benmoriceau it looks like this test is still mocking out several pieces of the secret migrator functionality. Don't you think it would be beneficial to create a real source and destination spec that have actual (test) secret values in them, rather than mocking out all of the pathing logic here?